mistletoe/block_token.py

```diff
@@ -182,10 +188,14 @@ class SetextHeading(BlockToken):
 
     Attributes:
         level (int): heading level.
+        children (list): inner tokens.
     """
     repr_attributes = ("level",)
+
     def __init__(self, lines):
-        self.level = 1 if lines.pop().lstrip().startswith('=') else 2
+        underline = lines.pop().strip()
```

Comment:
Sure, saving the whole line instead of `underline_length` does make things a little simpler. The `level` attribute is still needed by HtmlRenderer and others, though.